### PR TITLE
Add Gated-check-in and release CI

### PR DESCRIPTION
**Pull request by 928PJY**

### CODE REVIEW

File: src/test/unitTests/extension.unit.test.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { expect } from 'chai';
+
+describe('Extension Unit Test sample', () => {
+    it('Sample test', () => {
+        expect([1, 2, 3].indexOf(2)).to.be.equal(1);
```
Comment (yufeih) [inline src/test/unitTests/extension.unit.test.ts]: remove this sample or replace it with a more meaningful test #Resolved

File: azure-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  tags:
+    include:
+    - '*'
+
+strategy:
+  matrix:
+    mac:
+      imageName: 'macos-10.13'
+    windows:
+      imageName: 'vs2017-win2016'
```
Comment (yufeih) [inline azure-pipeline.yml]: does `windows-latest` and `mac-latest` work? #Resolved

File: azure-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+steps:
+
+- task: NodeTool@0
+  displayName: 'Use Node 10.x'
+  inputs:
+    versionSpec: 10.x
+
+- task: Npm@1
+  displayName: 'npm install'
```
Comment (yufeih) [inline azure-pipeline.yml:40]: can use https://docs.microsoft.com/en-us/azure/devops/pipelines/caching/?view=azure-devops to improve npm install perf on hosted agent #Resolved

**Review by yufeih — APPROVED** [review]

File: docs/RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Release Instruction
+
+Please follow the following steps to publish a new version extension.
+
+1. Update the `version` property in [package.json](../package.json) file.
+
+2. Update [CHANGELOG](../CHANGELOG.md) file to add the notable changes of this version.
```
Comment (live1206) [inline docs/RELEASE.md:7]: no date in [`CHANGELOG.md`](https://github.com/docascode/vscode-docs-build/blob/master/CHANGELOG.md)? #Resolved

File: azure-pipeline.yml (hunk 3 above)
Comment (928PJY) [reply azure-pipeline.yml:40]: Good point, that is exactly what I want.

---
In reply to: [361399894](https://github.com/docascode/vscode-docs-build/pull/6#discussion_r361399894) [](ancestors = 361399894)

File: docs/RELEASE.md (hunk 4 above)
Comment (928PJY) [reply docs/RELEASE.md:7]: Good point, Added

---
In reply to: [361413067](https://github.com/docascode/vscode-docs-build/pull/6#discussion_r361413067) [](ancestors = 361413067)

**Review by live1206 — APPROVED** [review]

**Pull request merged**